base/iostream.jl

```diff
@@ -47,17 +47,18 @@ macro _lock_ios(s, expr)
 end
 
 """
-    fd(stream)
+    fd(stream) -> RawFD
 
 Return the file descriptor backing the stream or file. Note that this function only applies
 to synchronous `File`'s and `IOStream`'s not to any of the asynchronous streams.
 
-File descriptors should typically be represented as [`RawFD`](@ref) objects, rather
-than as `Int`s, to ensure that they are properly interpreted by Julia functions.
+`RawFD` objects can be passed directly to other languages via the `ccall` interface.
 
-Note that `RawFD` objects can be passed directly to other languages via the `ccall` interface.
+!!! compat "Julia 1.12"
+    Prior to 1.12, this function returned an Int instead of a RawFD. You may use
+    `RawFD(fd(x))` for compatibility with earlier Julia versions.
```

Comment:
Personally I was slightly confused by the comment as it is written now. This may be entirely on me! But I think this would have been clearer to me (but feel free to disagree with me )
```suggestion
    `RawFD(fd(x))` in your code to stay compatible with earlier Julia versions.
```

Reply:
Perhaps this clarifies things?

```suggestion
    `RawFD(fd(x))` to produce a `RawFD` in all Julia versions.
```

Reply:
I think the crux of the issue was that "compatibility" is a vague term.

Reply:
Exactly! And so your suggestion works for me, too